CONTRIBUTING.md

```diff
@@ -0,0 +1,258 @@
+* writing tutorials, guides and other informal documentation and sharing it on [awesome-bevy](https://github.com/bevyengine/awesome-bevy)
+* streaming, writing blog posts about creating your game or creating videos. Share these in the `#devlogs` channel on Discord!
+
+### Writing plugins
+
+If you're interested in contributing to the community ecosystem in a way that doesn't make sense as part of the core engine,
+feel free to write a plugin or crate for Bevy!
+
+Any useful, reusable and non-trivial pieces of functionality that you've made serves as a good candidate for a plugin.
+If it's closer to a snippet or design pattern, you may want to share it with the community on Discord, Reddit or GitHub Discussions instead.
+
+Check out our [plugin guidelines](https://github.com/bevyengine/bevy/blob/main/docs/plugins_guidelines.md) for helpful tips and norms.
+
+### Fixing bugs
+
+Bugs in Bevy (or our website / books) are filed on the issue tracker corresponding to where they're found, using the [`bug`](https://github.com/bevyengine/bevy/issues?q=is%3Aissue+is%3Aopen+label%3Abug) label.
+
+If you're looking for an easy place to start, take a look at the [`good-first-issue`](https://github.com/bevyengine/bevy/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) tag, and feel free to ask questions on the issue in question or on Discord.
+You don't need anyone's permission to try fixing a bug or adding a simple feature, but stating that you'd like to tackle an issue can be helpful to avoid duplicated work.
+
+When you make a pull request that fixes an issue, include a line that says `Fixes #X` (or "Closes"), where `X` is the issue number.
+This will cause the issue in question to be closed when your PR is merged.
+
+General improvements to code quality are also welcome!
+Bevy can always be safer, better tested, and more idiomatic.
+
+### Writing docs
+
+Like every other large, rapidly developing open source library you've ever used, Bevy's documentation could stand to be improved.
+This is incredibly valuable, easily distributed work, but requires a bit of guidance:
+
+* inaccurate documentation is worse than no documentation: prioritize fixing broken docs
+* Bevy is remarkable unstable: before tackling a new major documentation project, check in with the community about the stability of that feature and upcoming plans to save yourself heartache
+* code documentation (doc examples and in the examples folder) is easier to maintain because the compiler will tell us when it breaks
+* inline documentation should be fairly direct, minimal and technical
+* the Bevy book is hosted on the `bevy-website` repo and targeted towards beginners who are just getting to know Bevy (and perhaps Rust!)
+* RFCs are not documentation: they serve as a record of accepted decisions
+
+[docs.rs](https://docs.rs/bevy) is built out of the last release's documentation, which is written right beside the adjacent code.
```

Comment:
~~built out of~~ built from

Reply:
~~written right beside the adjacent code.~~ written inline, above the corresponding code.